### PR TITLE
!str introduce StreamletLogic trait

### PR DESCRIPTION
**Pull request by hepin1989**
It would be better to share the same structure.

### CODE REVIEW

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletLogic.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.streamlets
+
+trait StreamletLogic[Context <: StreamletContext] extends Serializable {
+  def context: Context
+
+  def getContext(): Context = context
```
Comment (hepin1989) [inline core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletLogic.scala]: With this

**Comment by RayRoestenburg** [comment]
@hepin1989 thanks for the PR! I'll have a look soon, looks good in general. With the Java API, always good to add a Java test to see it works as expected.

File: core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletLogic.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletLogic.scala]: Should be good, yes.

**Comment by hepin1989** [comment]
@RayRoestenburg I am done with it

**Comment by RayRoestenburg** [comment]
@hepin1989 please run `sbt test` in root of project, there are still compilation errors, in the unit tests.
`cloudflow-akka-tests/src/test/scala/cloudflow/akkastream/scaladsl/AkkaStreamletSpec`, 
`cloudflow-flink-tests/src/test/java/cloudflow/flink/javadsl/FlinkProcessor.java`,

```
[error] (cloudflow-akka-tests / Test / compileIncremental) Compilation failed
[error] (cloudflow-flink-tests / Test / compileIncremental) javac returned non-zero exit code
```

**Comment by hepin1989** [comment]
@hepin1989 sorry,I pushed to the wrong branch

**Pull request closed without merging**